PyStemmusScope/iostreamer.py

```diff
@@ -0,0 +1,149 @@
+
+        Args:
+            path_to_config_file: Path to the config file.
+
+        Returns:
+            Dictionary containing paths to work directory and all sub-directories.
+        """
+        config = {}
+        with open(path_to_config_file, "r", encoding="utf8") as f:
+            for line in f:
+                (key, val) = line.split("=")
+                config[key] = val.rstrip('\n')
+
+        return config
+
+    def prepare_work_dir(self, forcing_filenames_list: list,
+        full_run: bool = False
+    ):
+        """Prepare work directory for each station.
+
+        Prepare work directory based on the given list of forcing. This function
+        creates directory for each forcing data and name it using the name of the
+        station.
+
+        Args:
+            forcing_filenames_list: List of forcing filenames to be used for the experiments.
+            full_run: If true, all stations with forcing listed in the ForcingPath will be used.
+
+        Returns:
+            Dictionary containing path to work directory and config file for every station/forcing.
+        """
+        self.forcing_filenames_list = forcing_filenames_list
+        
+        if full_run:
+            self.forcing_filenames_list = [file.name for file in Path(self.config["ForcingPath"]).iterdir()]
+
+        # dict to store path to config file and work directory for each station
+        self.work_dir_dict, self.config_path_dict = self._create_work_dir()
+
+        return self.work_dir_dict, self.config_path_dict
+
+    def _create_work_dir(self):
+        """Create input directory and copy forcing files.
+
+        Work flow executor to create work directory and all sub-directories.
+
+        Returns:
+            Dictionary containing path to work directory and config file for every station/forcing.
+        """
+        # empty dict to store path to config file and work directory for each station
+        config_path_dict = {}
+        work_dir_dict = {}
+        for ncfile in self.forcing_filenames_list:
+            # get start time with the format Y-M-D-HM
+            timestamp = time.strftime('%Y%m%d_%H%M')
+            station_name = ncfile.split('_')[0]
+            # create input directory
+            work_dir = Path(self.config["InputPath"], station_name + '_' + timestamp)
+            Path(work_dir).mkdir(parents=True, exist_ok=True)
+            print(f"Prepare work directory {work_dir} for the station: {station_name}")
+            # copy model parameters to work directory
+            self._copy_data(work_dir)
+            # update config file for ForcingFileName and InputPath
+            config_file_path = self._update_config_file(ncfile, work_dir, station_name, timestamp)
+            # save config path and work directory to the dictionaries
+            work_dir_dict[ncfile] = work_dir
+            config_path_dict[ncfile] = config_file_path
+
+        return work_dir_dict, config_path_dict
```

Comment:
```suggestion

        # get start time with the format Y-M-D-HM
        timestamp = time.strftime('%Y%m%d_%H%M')
        station_name = ncfile.split('_')[0]

        # create input directory
        input_dir = Path(f"{self.config['WorkDir']}/input/{station_name}_{timestamp}")
        input_dir.mkdir(parents=True, exist_ok=True)
        message = f"Prepare work directory {input_dir} for the station: {station_name}"
        logger.info("%s", message)
        # copy model parameters to work directory
        self._copy_data(input_dir)
        self.input_dir = str(input_dir)

        # create output directory
        output_dir = Path(f"{self.config['WorkDir']}/output/{station_name}_{timestamp}")
        output_dir.mkdir(parents=True, exist_ok=True)
        message = f"Prepare work directory {output_dir} for the station: {station_name}"
        logger.info("%s", message)
        self.output_dir = str(output_dir)

        # update config file for ForcingFileName and InputPath
        self.config_file_path = self._update_config_file(ncfile, self.input_dir, self.output_dir, station_name, timestamp)

        return self.input_dir, self.output_dir, self.config_file_path
```

Reply:
As discussed, we need to create input and output directories. Also, no need to have the loop `for ncfile in self.forcing_filenames_list` inside the class functions. We can implement it inside the notebook for more transparency.  As you can see, I provided some suggestions: 

- logger instead of print statement in python modules. 
- always functions return paths as string and not path objects.
- `f""` instead of using `+` in string concatenation.

Reply:
Thanks for the suggestions and the refactoring. It is revised.